gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java

```diff
@@ -795,25 +837,69 @@ public void runOldEntriesCleanerThread(){
 
   class OldEntriesCleanerThread implements Runnable {
     // Keep each entry alive for at least 20 secs.
+
     public void run() {
       try {
         if (!oldEntryMap.isEmpty()) {
+          // Can't do map.clear as have to account for memory for each oldEntry
+          if (getTxManager().getHostedTransactionsInProgress().size() == 0) {
+            acquireWriteLockOnSnapshotRvv();
+            try {
+              if (getTxManager().getHostedTransactionsInProgress().size() == 0) {
+                if (getLoggerI18n().fineEnabled()) {
+                  getLoggerI18n().info(LocalizedStrings.DEBUG, "Clearing the Map");
+                }
+                for (Entry<String, Map<Object, BlockingQueue<RegionEntry>>> entry : oldEntryMap.entrySet()) {
+                  Map<Object, BlockingQueue<RegionEntry>> regionEntryMap = entry.getValue();
+                  LocalRegion region = (LocalRegion)getRegion(entry.getKey());
+                  for (Entry<Object, BlockingQueue<RegionEntry>> oldEntry : regionEntryMap.entrySet()) {
+                    for (RegionEntry re : oldEntry.getValue()) {
```

Comment:
Is it necessary to iterate the whole map once again? This is getting to be very expensive and will likely have a significant hit. We need to fix this asap. Please file a high priority task.

Reply:
Had to account for memory for each of the entry being deleted.